great_expectations/data_context/data_context.py

```diff
@@ -3259,6 +3263,32 @@ def run_checkpoint(
             **kwargs,
         )
 
+    def add_profiler(
+        self,
+        name: str,
+        config_version: float,
+        rules: Dict[str, dict],
+        class_name: str,
+        module_name: Optional[str] = None,
+        variables: Optional[dict] = None,
+        commented_map: Optional[CommentedMap] = None,
+        ge_cloud_id: Optional[str] = None,
+    ):
+        config: RuleBasedProfilerConfig = ruleBasedProfilerConfigSchema.load(
+            {
+                "name": name,
+                "config_version": config_version,
+                "rules": rules,
+                "class_name": class_name,
+                "module_name": module_name,
+                "variables": variables,
+                "commented_map": commented_map,
+            }
+        )
+        return profiler_toolkit.add_profiler(
+            config=config, data_context=self, ge_cloud_id=ge_cloud_id
+        )
```

Comment:
Should be `RuleBasedProfiler.add_profiler` class method but putting everything in toolkit and saving all the updates for a refactoring PR.

Reply:
@cdkini I think your approach is perfect for today's needs.

Reply:
can we take data_context out?